Monorepo

chrisprice opened the pull request:
Such trend :+1: 

Whilst this looks like a whooper, the last 2 commits are the significant ones.

The rest of the commits were imported using `lerna import`

Problems -

* [ ] Looks like d3fc-label-layout could cause some issues as it has its own site
* [x] Travis seems to have found a problem with the yarn workspaces config -
> yarn install v1.3.2
> error Workspaces can only be enabled in private projects
> info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
* [ ] `d3fc-technical-indicator` is causing a cyclic dependency

alisd23 committed:
Added ohlc and bar files

alisd23 committed:
Add minified files

alisd23 committed:
Fix exports

alisd23 committed:
Exported function now takes context

alisd23 committed:
Fixed context issue

alisd23 committed:
Add rest of the chart types

alisd23 committed:
Added tests

alisd23 committed:
Add travis.yml

alisd23 committed:
Don't commit build artifacts --tags

alisd23 committed:
Add eslint

alisd23 committed:
Fix line endings issue

alisd23 committed:
Housekeeping #4

alisd23 committed:
Exported factory functions accept context

alisd23 committed:
Add readme

chrisprice committed:
Change description

chrisprice committed:
Update description

chrisprice committed:
Standardise the API documentation format

chrisprice committed:
0.0.2

chrisprice committed:
Remove d3 dependency

chrisprice committed:
Add build to npm package

chrisprice committed:
feat: Integrate semantic release

chrisprice committed:
chore: Fix travis build

chrisprice committed:
fix: No version field in package.json

alisd23 committed:
test: Add tests for different options

alisd23 committed:
docs: Improve readme

alisd23 committed:
feat: Rename exported UMD module to fc_path

BREAKING CHANGE: Previously the module was called fcPath. The new name
better reflects the d3v4 naming conventions.

alisd23 committed:
fix: Correct contexts

alisd23 committed:
chore: change packge name to d3fc-shape

BREAKING CHANGE: Previously called d3fc-path

alisd23 committed:
test: Fix tests

alisd23 committed:
chore: remove version from package.json

chrisprice committed:
docs: Update README to latest version

BREAKING CHANGE: The package has been renamed to d3fc-shape.

chrisprice committed:
docs: Include ctx.stroke() in canvas example

chrisprice committed:
refactor: context now passed as a property

Fixes #7

BREAKING CHANGE: The context can no longer be supplied as an argument
to the factory, instead it must be specified as a property. E.g.
```js
fc.bar().context(canvas.getContext())
```
The implementation defaults to using d3-path if no context is specified.

alisd23 committed:
docs: Fix docs example code

chrisprice committed:
feat: Initial version of rebind, rebindAs & rebindAll

chrisprice committed:
docs: Add API documentation

chrisprice committed:
test: Remove duplicate test

chrisprice committed:
docs: Add link to d3.rebind

chrisprice committed:
fix: Remove unnecessary escaping

chrisprice committed:
chore: chore: Move eslint global to test folder

chrisprice committed:
feat: simplify API by using explicit transforms

BREAKING CHANGE: changed `rebindAll` signature
`rebindAll(target, source, prefix, exclusions/...exclusions)` is now
`rebindAll(target, source, ...transforms)` where each transform is a
function controlling if and how source properties are mapped to target
properties. It accepts the source property name and returns the name to
use on the target or any falsey value to indicate it should not be copied
over.

As well as creating transforms manually, you can also make use of -

* `exclude(...names)` - exclude a set of property names or regexes
* `include(...names)` - include only the set of property names or regexes
* `map({ targetName: sourceName, ... })` - map the specified properties
* `prefix(str)` - prefix all property names with str

Transforms can be chained e.g. `exclude('foo').prefix('bar')` but be
Be aware that transforms are evaluated left to right, so e.g.
`prefix('bar'), exclude('foo')` will never exclude `foo` because it does
not start with the prefix `bar`.

BREAKING CHANGE: removed `rebindAs`
Use `rebindAll(target, source, map({ 'foo': 'bar' }))`

chrisprice committed:
docs: fix map docs, swap source/target

chrisprice committed:
chore: remove unused file

chrisprice committed:
docs: break out transform methods and fix link

ColinEberhardt committed:
docs: formatting change

Also provides a bit of structure to separate the primary API from 'utility' functions

ColinEberhardt committed:
refactor: simple name change

chrisprice committed:
refactor: rename map to includeMap

BREAKING CHANGE: `map` is renamed `includeMap`. This was done to make it
more obvious that any properties not specified as keys will be implicitly
excluded.

chrisprice committed:
chore: fix styling

ColinEberhardt committed:
docs: standardised the prefix

ColinEberhardt committed:
doc: standardising prefix and format

ColinEberhardt committed:
docs: consistent formatting

Made the formatting more consistent with the other components. Also, I think down-sampling is a more common term than sub-sampling.

alisd23 committed:
fix: property accessors - closes #11

Commit:
feat: Initial d3fc-sample implementation

tlsim committed:
fix: Prevent excluding the built js by adding empty .npmignore (#4)

tlsim committed:
feat: Initial commit - migrated from d3fc, ES2015-ify

docs: Add API reference

ColinEberhardt committed:
docs: minor documentation improvements, adding images

jleft committed:
fix: rebind and rebindAll now return target

jleft committed:
test: add tests to check the target is returned by rebind and rebindAll

tlsim committed:
docs: corrected rebind and rebindAll signatures

swap order of source and target.

tlsim committed:
chore: set up semantic release

ColinEberhardt committed:
Merge pull request #5 from d3fc/fixes

Fixes

ColinEberhardt committed:
Merge pull request #6 from ColinEberhardt/excel-tests

using excel-generated test data

tlsim committed:
refactor: use d3-array and d3fc-rebind

BREAKING CHANGE: d3fc-sample now depends on d3-array and d3fc-rebind. It
no longer depends on d3.

ColinEberhardt committed:
Merge pull request #13 from d3fc/tests

test: update all specs to use CSV data

ColinEberhardt committed:
Merge pull request #12 from d3fc/api-consistency

Fixes

ColinEberhardt committed:
Merge pull request #14 from d3fc/move-design-docs-to-readme

Move design docs to readme

chrisprice committed:
Merge pull request #15 from d3fc/null-handling

Null handling

tlsim committed:
docs: Add undefined input behaviour to README (#16)

tlsim committed:
refactor: RSI: abstract out average calculation (#17)

Calculate changes using [prevClose, close] pairs

tlsim committed:
feat: initial commit - migrate from d3fc, ES2015-ify

tlsim committed:
fix: Add TypeScript definition file (#18)

ColinEberhardt committed:
feat: changed global name from fc_rebind to simply fc

BREAKING CHANGE: references to the global variable fc_rebind should be replaced with a reference to fc

ColinEberhardt committed:
chore: removed bundle step from npm test

ColinEberhardt committed:
Merge pull request #19 from ColinEberhardt/master

changed global name

ColinEberhardt committed:
feat: changed global name from fc_sample to simply fc

BREAKING CHANGE: references to the global variable fc_sample should be replaced with a reference to fc

ColinEberhardt committed:
chore: test no longer depends on bundling

ColinEberhardt committed:
fix: the current code was failing the no-return-assign rule

ColinEberhardt committed:
feat: changed global name from fc_shape to fc …

BREAKING CHANGE: references to the global fc_shape object need to be updated to fc

ColinEberhardt committed:
chore: remove the bundle step from the unit test

ColinEberhardt committed:
fix: added missing dependencies (and remove unused ones)

fixed minor listing issue

ColinEberhardt committed:
feat: changed global name from fc_random_data to simply fc

BREAKING CHANGE: references to the global variable fc_random_data should be replaced with a reference to fc

ColinEberhardt committed:
test: ensure bundle is also unit tested

also updated to non-beta versions of d3 dependencies

ColinEberhardt committed:
fix: bundle before release

Previously this happened as part of npm test, do happened as a side-effect

ColinEberhardt committed:
fix: npm *run* bundle (#10)

FFS

greenkeeperio-bot committed:
chore(package): update dependencies (#11)

https://greenkeeper.io/

ColinEberhardt committed:
chore: bundle should occur before npm publish

ColinEberhardt committed:
test: added bundle test and updated dependencies

ColinEberhardt committed:
docs: updated to use ES6 style

ColinEberhardt committed:
feat: namespace flattened

BREAKING CHANGE: All components exposed by this package are now prefixed with `random`

ColinEberhardt committed:
test: added a test to verify the bundle

ColinEberhardt committed:
chore: enabled semantic release (#7)

greenkeeperio-bot committed:
chore(package): update dependencies (#8)

https://greenkeeper.io/

ColinEberhardt committed:
chore: changed d3fc dependency to d3fc-random-data

ColinEberhardt committed:
feat: renaming to reflect namespace flattening

BREAKING CHANGE: All exposed components are now prefixed with `shape`, e.g. `bar` becomes `shapeBar`

ColinEberhardt committed:
chore: moved to latest version of d3-path

greenkeeperio-bot committed:
chore(package): update dependencies (#15)

https://greenkeeper.io/

ColinEberhardt committed:
fix: site demo now works with D3v4 (#11)

greenkeeperio-bot committed:
chore(package): update eslint to version 3.1.0 (#12)

https://greenkeeper.io/

greenkeeperio-bot committed:
chore(package): update eslint to version 3.1.0 (#22)

https://greenkeeper.io/

ColinEberhardt committed:
chore: created from archetype

ColinEberhardt committed:
feat: Created first release of d3fc-series.

Major changes:
 + Supports Canvas as well as SVG series
 + When the series orientation changes the scales are transposed but not the value accessors
 + Value accessors have been renamed for consistency

ColinEberhardt committed:
Merge pull request #20 from ColinEberhardt/master

Various fixes

greenkeeperio-bot committed:
chore(package): update d3-dsv to version 1.0.1 (#25)

https://greenkeeper.io/

greenkeeperio-bot committed:
chore(package): update rollup to version 0.34.3 (#15)

https://greenkeeper.io/

ColinEberhardt committed:
chore: removed un-used dependency (#26)

greenkeeperio-bot committed:
chore(package): update rollup to version 0.34.5 (#32)

https://greenkeeper.io/

ColinEberhardt committed:
chore: removed eslint-plugin-promise (#33)

chrisprice committed:
Initial commit

chrisprice committed:
Initial implementation

chrisprice committed:
feat: add test/helpers/* to jasmine helpers

chrisprice committed:
1.1.0

chrisprice committed:
fix: test task was not bundling

chrisprice committed:
1.1.1

chrisprice committed:
fix: test task not waiting for bundle to complete

chrisprice committed:
1.1.2

chrisprice committed:
fix: globals not being configured correctly

chrisprice committed:
1.1.3

chrisprice committed:
1.1.4

chrisprice committed:
fix: bin script failing due to crlf

chrisprice committed:
feat: integrate semantic release

Adds it as a dependency so other modules don't need it (rather than a devDependency like normal)

chrisprice committed:
fix: add no-op test script

chrisprice committed:
fix: run semantic release on success

chrisprice committed:
chore: integrate d3fc-scripts (#13)

chrisprice committed:
chore: integrate d3fc-scripts (#34)

chrisprice committed:
chore: integrate d3fc-scripts (#20)

chrisprice committed:
chore: integrate d3fc-scripts (#21)

chrisprice committed:
chore: integrate d3fc-scripts (#15)

chrisprice committed:
refactor: remove duplicated code

chrisprice committed:
chore: update to es2015

chrisprice committed:
feat: rebind now supports properties found on the prototype

Closes #12.
Fixes #13.

alisd23 committed:
docs: make readme format consistent (#22)

* docs: make readme format consistent

* docs: Add API contents

* docs: Add missing links

* docs: add bucket link

alisd23 committed:
docs: make readme format consistent (#35)

alisd23 committed:
docs: make readme format consistent (#21)

ColinEberhardt committed:
chore: added semantic release (#2)

greenkeeperio-bot committed:
chore(package): update dependencies (#3)

https://greenkeeper.io/

alisd23 committed:
docs: make readme format consistent (#14)

alisd23 committed:
docs: make readme format consistent (#17)

ColinEberhardt committed:
fix: added correct module entry points (#7)

fixes #6

alisd23 committed:
docs: make readme format consistent (#5)

chrisprice committed:
docs: remove reference to archetype-project

chrisprice committed:
fix: catch unhandled promise error

alisd23 committed:
feat: add any d3 or d3fc lib to rollup globals

alisd23 committed:
feat: add node resolution

alisd23 committed:
feat: add examples folder to eslint glob (#18)

greenkeeperio-bot committed:
chore(package): update rollup to version 0.35.11 (#17)

https://greenkeeper.io/

greenkeeperio-bot committed:
chore(package): update babel-preset-es2015 to version 6.14.0 (#3)

https://greenkeeper.io/

alisd23 committed:
test: add test result output

greenkeeperio-bot committed:
chore(package): update eslint-config-standard to version 6.0.1

https://greenkeeper.io/

greenkeeperio-bot committed:
chore(package): update rollup to version 0.36.0

https://greenkeeper.io/

alisd23 committed:
fix: add missing dependency (#22)

alisd23 committed:
chore: fix lint error

chrisprice committed:
fix: building on windows

root cause: https://github.com/developit/modify-babel-preset/pull/10
babel-preset-es2015-rollup not yet updated: https://github.com/rollup/babel-preset-es2015-rollup/issues/11

chrisprice committed:
docs: typo

ColinEberhardt committed:
fix: bar-baseline is now correctly computed (#14)

chrisprice committed:
chore: enable semantic release

chrisprice committed:
fix: build folder not published

alisd23 committed:
feat: add required files check to test (#23)

chrisprice committed:
Merge pull request #15 from ColinEberhardt/decorate-docs

Decorate docs

ColinEberhardt committed:
fix: added support for decorate with the SVG multi-series

alisd23 committed:
fix: required files check not running on every file

alisd23 committed:
feat: add screenshot task (#25)

alisd23 committed:
fix: fix test (#27)

greenkeeperio-bot committed:
chore(package): update d3fc-scripts to version 1.6.1

https://greenkeeper.io/

greenkeeperio-bot committed:
chore(package): update d3fc-scripts to version 1.6.1

https://greenkeeper.io/

chrisprice committed:
fix: add required files

chrisprice committed:
fix: add required files

chrisprice committed:
fix: add required files

chrisprice committed:
fix: add required files

chrisprice committed:
fix: add required files

chrisprice committed:
fix: add required files

chrisprice committed:
fix: add required files

chrisprice committed:
fix: add required files

chrisprice committed:
fix: add required files

greenkeeperio-bot committed:
chore(package): update d3fc-scripts to version 1.6.1 (#17)

https://greenkeeper.io/

greenkeeperio-bot committed:
chore(package): update d3fc-scripts to version 1.6.1 (#25)

https://greenkeeper.io/

greenkeeperio-bot committed:
chore(package): update d3fc-scripts to version 1.6.1 (#40)

https://greenkeeper.io/

greenkeeperio-bot committed:
chore(package): update d3fc-scripts to version 1.6.1 (#25)

https://greenkeeper.io/

greenkeeperio-bot committed:
chore(package): update d3fc-scripts to version 1.6.1 (#20)

https://greenkeeper.io/

alisd23 committed:
fix: file comparison (#28)

jleft committed:
docs: correct package name in installation instructions (#6)

alisd23 committed:
Merge pull request #37 from alisd23/add-examples

feat: add examples

ColinEberhardt committed:
feat: allow decoration of canvas multi series

alisd23 committed:
docs: add styles to canvas example (#42)

greenkeeperio-bot committed:
chore(package): update d3fc-data-join to version 3.0.5 (#25)

https://greenkeeper.io/

ColinEberhardt committed:
Merge pull request #24 from ColinEberhardt/group-series-squash

feat: Added grouped series and documented how to render stacked series

greenkeeperio-bot committed:
chore(package): update topojson to version 2.0.0 (#28)

https://greenkeeper.io/

greenkeeperio-bot committed:
chore(package): update fs-promise to version 1.0.0 (#43)

https://greenkeeper.io/

ColinEberhardt committed:
fix: transitions are now propagated (#29)

ColinEberhardt committed:
Merge pull request #32 from chrisprice/master

fix: transition support in multi

ColinEberhardt committed:
fix: propagate transitions to the inner selection

greenkeeperio-bot committed:
chore(package): update d3-path to version 1.0.3

https://greenkeeper.io/

chrisprice committed:
chore: fix indentation lint failure (#20)

chrisprice committed:
feat: implement the iterable protocol (#4)

* chore: fix indentation lint failure

* feat: implement the iterable protocol

Fix #1

ColinEberhardt committed:
fix: propagate transitions to all selections

Also, ensure that the enter selection sets the container translation so that the error bars / box plots don't transition in from the origin.

ColinEberhardt committed:
feat: updated the multi-series mapping function to be consistent with selection.each

BREAKING CHANGE: The signature of the multi-series mapping function has changed from (data, dataSeries, index), to (data, index, series)

fixes #35
improved documentation for multi-series fixes #34

greenkeeperio-bot committed:
chore(package): update d3fc-series to version 2.0.1 (#45)

https://greenkeeper.io/

jleft committed:
docs: correct bucket documentation (#30)

jleft committed:
test: add tests for bucket utility (#32)

mbssantos committed:
feat: add 'lint' to d3fc-scripts (#45)

jleft committed:
fix: bucket utility now returns the same number of buckets as the data length if the bucket size is less than 1

jleft committed:
test: add test for bucket utility when bucket size is less than 1

jleft committed:
test: add test for bucket utility when bucket size is less than 1

jleft committed:
fix: correctly expose on

jleft committed:
fix: correct usage of d3-dispatch

jleft committed:
test: add bundle spec

jleft committed:
docs: correct event name from pointer to point

jleft committed:
test: add pointer tests

greenkeeperio-bot committed:
chore(package): update d3fc to version 12.0.0 (#46)

https://greenkeeper.io/

jleft committed:
docs: fix decorate pattern link (#40)

ColinEberhardt committed:
chore: update all dependencies (#49)

chrisprice committed:
chore: tidy up dependencies

chrisprice committed:
fix: custom css and multi.mapping signature

WPFerg committed:
fix: allow 2 dimensional data arrays

WPFerg committed:
test: add tests for 2-dimensional data arrays

ColinEberhardt committed:
feat: adding a repeat series (#42)

Commit:
Fix code samples in README.md (#21)

j-n-white committed:
docs: Fixed documentation for seriesRepeat Orient (#45)

j-n-white committed:
docs: Clarify that seriesPoint.size sets area (#47)

ColinEberhardt committed:
feat: improved series bar width concepts

fixes #44 #26

BREAKING CHANGE: fractionalBarWidth has been replaced with autoBandwidth

The previous fractionalBarWidth concept was problematic in that it required the series to supply all of the data to the barWidth property (function) in order to support its behaviour. This was not good from a performance perspective, and inflexible.

Following this change, series now expose a bandwidth property (which reflects the same property exposed by d3 band / point scales). This property specifies the width in screen values (pixels).

For example, to create bars with a width of 50 pixels:

```
var svgBar = fc.seriesSvgBar()
    .crossValue(function(d) { return d.x; })
    .mainValue(function(d) { return d.y; })
    .width(50);
```

You can also provide a function that allows configuration of the bandwidth on a per-datapoint basis:

```
var svgBar = fc.seriesSvgBar()
    .crossValue(function(d) { return d.x; })
    .mainValue(function(d) { return d.y; })
    .width(function(d, i) { return i % 2 * 10; });
```

In order to automatically compute the width based on the space between datapoints, you can adapt a series with the autoBandwith component:

```
var svgBar = fc.autoBandwidth(fc.seriesSvgBar())
    .crossValue(function(d) { return d.x; })
    .mainValue(function(d) { return d.y; })
    .widthFraction(0.5);
```

This is equivalent to the previous code:

```
var svgBar = fc.seriesSvgBar()
    .crossValue(function(d) { return d.x; })
    .mainValue(function(d) { return d.y; })
    .barWidth(fc.fractionalBarWidth(0.5));
```

Finally, the autoBandwidth, when used in conjunction with scales that expose a bandwidth property (band / point), this value is propagated to the series.

ColinEberhardt committed:
docs: document use of auto bandwidth with multi series (#51)

fixes #50

ColinEberhardt committed:
Update README.md (#53)

ColinEberhardt committed:
Merge pull request #43 from ColinEberhardt/heatmap-series

Heatmap series

ColinEberhardt committed:
fix: ensure the default styles are applied on the enter selection (#55)

Commit:
fix: use shoelace-style area calculation algorithm

Commit:
feat: rename randomGeometricBrownianMotion randomNormal to random

Although the default function used by randomGeometricBrownianMotion to generate random numbers is d3_random.randomNormal, it does not have to be a normal distribution

BREAKING CHANGE: randomGeometricBrownianMotion.randomNormal([value]) renamed randomGeometricBrownianMotion.random([value])

Commit:
feat: randomFinancial exposes random function used by the geometric Brownian motion simulation

Closes #23

Commit:
feat: allow base.defined to be overwritten

All base files have been refactored to use a base constructor which adds getters and setters based on the properties passed to it

Closes #57

Commit:
fix: allow xScale and yScale to be overwritten

Commit:
fix: remove unused import in src/base.js

Commit:
fix: add default seed value of 0 to geometric Brownian motion

Fixes #9

msuperina committed:
fix: use base.xScale() to invoke current xScale function (#61)

Closes #60

msuperina committed:
refactor: cache xScale and yScale functions when used multiple times in the same scope (#62)

test: ensure xScale and yScale can be called and overridden

ColinEberhardt commented:
We should make this one a priority ... we have just about finished a new release of d3fc, and semantic-release is breaking for a number of packages. Firstly due to the revoked npm tokens, and secondly, lots of our GH tokens are not working either - possibly manually revoked by whoever originally set up semantic-release for these packages: https://github.com/d3fc/d3fc/pull/1103

chrisprice committed:
chore: fix cross-dependency versions

chrisprice committed:
chore: move site into packages

chrisprice committed:
chore: fix tests and disable failing test

chrisprice committed:
chore: disable d3fc-scripts

chrisprice committed:
chore: fix example script references

chrisprice committed:
fix: d3.csv/json breaking changes

chrisprice committed:
fix: no jsnext:main causing bundle error

chrisprice committed:
chore: bump to latest version of d3fc-scripts

chrisprice committed:
fix: externalise custom element polyfill

BREAKING CHANGE: The custom elements polyfill is no longer bundled, see the d3fc-element README for more details on how to explicitly include it if required

chrisprice committed:
chore: pull d3fc-scripts dependencies to top level

chrisprice committed:
chore: fix dist/build issues

chrisprice committed:
fix: don't transpile classes for custom elements

chrisprice committed:
chore: fix d3fc bundle when running test

chrisprice committed:
chore: build site as default task

chrisprice committed:
chore: remove semantic release

chrisprice committed:
chore: rename all packages @d3fc/d3fc-

chrisprice committed:
chore: update travis build

chrisprice committed:
docs: fix link to package repository
